openedx/core/djangoapps/discussions/docs/decisions/0008-reported-content-email-notifications.rst

```diff
@@ -0,0 +1,77 @@
+Reported content email notifications for moderators
+===================================================
+
+
+Status
+------
+
+Proposal
+
+
+Context
+-------
+
+We intend to provider an option for quick visibility of reported content for
+moderators. We intend to implement this by building an option to send out an
+email to discussion moderators when a content (post/response/comment) is
+reported. This email will be sent out when a content is reported, and stays
+reported for the next 2 minutes.
+
+We intend to send the email to all roles that have discussion
+moderation privileges, namely Discussion Admin, Discussion Moderator,
+Community TA and Group Community TA.
+
+Group community TAs will receive emails for reported content in their own
+cohort.
+
+A toggle for this setting will be shown on the edX discussions configuration
+page. This toggle can be turned on or off at any point in time during the
+course run.
+
+
+Requirements
+------------
+
+We need to have the ability to enable/disable email notifications for reported
+content for a course.
+
+When course content is reported, we have to send out an email to moderators 2
+minutes after the content is flagged for moderation.
+
+
+Decision
+--------
+
+We need to have a toggle setting for courses to have the option to configure
+the email notifications for moderators. To achieve this, we will add a new
+field named `reported_content_email_notifications` to the
+`CourseDiscussionSettings` model and the default value for the field will be
+`False`. Course Team will have the option to toggle this through the
+discussions configuration page with the help of the discussion settings
+endpoint.
+
+Furthermore, we also need to handle sending out the email notification to the
+moderators 2 minutes after the content was reported. To handle this, we will
+add a Celery task with a delay of 120 seconds. The task will be responsible of
```

Comment:
The task will first check if the content is still in the reported state, and then send an email to moderators.